File: Problem-Solving/LeetCode/0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int len = intervals.size();
        if (len == 0) {
            return vector<vector<int>> {newInterval};
        }
        vector<vector<int> > output;
        int idx = 0;

        for (;idx < len && intervals[idx][1] < newInterval[0];idx++) {
            output.push_back(intervals[idx]);
        }

        for(;idx < len && intervals[idx][0] <= newInterval[1];idx++) {
            newInterval[0] = min(intervals[idx][0], newInterval[0]);
            newInterval[1] = max(intervals[idx][1], newInterval[1]);
        }
        output.push_back(newInterval);

        for(;idx < len;idx++) {
            output.push_back(intervals[idx]);
        }

        return output;
    }
};

File: Problem-Solving/LeetCode/0057-insert-interval/0057-insert-interval.go
func insert(intervals [][]int, newInterval []int) [][]int {
    n := len(intervals)
    if n == 0 {
        return [][]int{newInterval}
    }
    
    output := make([][]int, 0)
    idx := 0

    for ;idx < n && intervals[idx][1] < newInterval[0];idx++ {
        output = append(output, intervals[idx])  
    }
    
    for ;idx < n && newInterval[1] >= intervals[idx][0] ;idx++ {
        newInterval[0] = min(intervals[idx][0], newInterval[0])
        newInterval[1] = max(intervals[idx][1], newInterval[1])
    }
    output = append(output, newInterval)

    for ;idx < n; idx++ {
        output = append(output, intervals[idx])
    }

    return output
}